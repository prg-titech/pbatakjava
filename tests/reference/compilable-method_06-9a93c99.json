{
    "basename": "compilable-method_06-9a93c99",
    "cmd": "{infile}",
    "infile": "tests/method_06/build.sh",
    "infile_hash": "e17e83ade5919786f677ede999db4e49337d4dbfb47f2e2e25bc5bbf",
    "stdout": "compilable-method_06-9a93c99.stdout",
    "stdout_hash": "a899a6d88af05990f6b11ef4d25c88a9d8048de818a155522cc2c5e5",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}