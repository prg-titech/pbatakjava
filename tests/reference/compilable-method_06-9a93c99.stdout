
public interface Point {
  public int getX();

  public int getX(int z);

  public float getX__float();

  public float useGetX();
}

public interface Point_Factory {
  public Point make(int x, int y);
}

public class Point_v1 implements Point {
  public int x;

  public int y;

  public Point_v1(int x, int y) {
    this.x = x;
    this.y = y;
  }

  public int getX() {
    return this.x;
  }

  public int getX(int z) {
    return 0;
  }

  public float getX__float() {
    return 0;
  }

  public float useGetX() {
    return 0;
  }
}

public class Point_v1_Factory implements Point_Factory {
  public Point make(int x, int y) {
    return new Point_v1(x, y);
  }
}

public class Point_v2 implements Point {
  public float x;

  public float y;

  public Point_v2(int x, int y) {
    this.x = x;
    this.y = y;
  }

  public int getX(int z) {
    return z;
  }

  public float getX__float() {
    return this.x;
  }

  public float useGetX() {
    return this.getX__float();
  }

  public int getX() {
    return 0;
  }
}

public class Point_v2_Factory implements Point_Factory {
  public Point make(int x, int y) {
    return new Point_v2(x, y);
  }
}
